ci-scripts/test_agent/FirebaseClient.py

```diff
@@ -32,6 +33,8 @@ def __init__(self):
         )
         # Get a reference to the database service
         self.db = self.firebase.database()
+        # token refresh timer
+        self.RefreshTokenTime = 1800
```

Comment:
Sorry if I was not clear. I suggest scheduling the token refresh here and not in main.py
main.py does not need to be aware of this. Unless this causes a multi threading race issue. does it?

Reply:
oh i see, sorry for back and forth. i think i cover that now.